README.md

```diff
@@ -1,43 +1,160 @@
+
+### Creating the Class Variable
+
+Let's take a step back and think about the concept of responsibility. Whose job is it to know about *every instance of the `Song` class*? We have two choices right now: an instance of the song class or the `Song` class itself.
+
+It is not the responsibility of an individual song to know about all of the other songs. Keeping track of all of the songs that it creates, however, fits right into the purview of the Song class. 
+
+So, how can we tell the `Song` class to keep track of every instance that it creates? We use a class variable. 
+
+Let's create a class variable, `@@all`, that will store every instance of the `Song` class:
+
+```ruby
+class Song
+	
+	@@all = []
+	
+	attr_accessor :name
+
+	def initialize(name)
+		@name = name
+	end
+end
+```
+
+Notice that we set our class variable equal to an empty array. Arrays are perfect for storing lists of data, so we'll use an array to store our lists of `Song` instances. 
+
+Now that our class is *set up* to store the instances that it produces, we have to ask: *how* does it store these instances?
+
+### Adding Instances to the `@@all` Array
+
+Before we can answer this question, we should ask another. *When* should the `Song` class become aware of, or store, an instance of itself? 
+
+This should happen at the time on instantiation––when a new song gets created, it should be immediately stored by our `Song` class' `@@all`class variable. 
+
+We can implement this by simply adding the new instance that gets created into the array stored in `@@all` *inside our `#initialize` method.*
+
+Let's take a look:
+
+```ruby
+class Song
+	
+	@@all = []
+	
+	attr_accessor :name
+
+	def initialize(name)
+		@name = name
+		@@all << self
+	end
+end
+```
+
+In `#initialize` we use the `self` keyword to refer to the new object that has just been created by `#new`. Remember that when `#new` is called, it creates a new instance of the class and then calls `#initialize` on that new instance. So, `#initialize` is technically an instance method. Inside an instance method we are in what is called **method scope** and `self` will refer to whichever instance the method is being called on. 
+
+We push `self` into the array that is stored in `@@all`. In this way, the `@@all` class variable will point to an ever-growing array that contains every instance of the `Song` class that gets created. 
+
+### Our Code in Action
+
+Let's see what happens when we actually execute the code we've written:
+
+```ruby
+99_problems = Song.new("99 Problems")
+thriller = Song.new("Thriller")
+```
+
+Now that we've created some songs, let's ask our `Song` class to show us all of the instances that we just created:
+
+```ruby
+Song.all
+  => NoMethodError: undefined method `all' for Song:Class
+```
+
+Uh-oh, looks like we don't have a class method to access the contents of the `@@all` array. Let's build one now:
```

Comment:
append."just like instance variables, we need to create a method for our class variable to be accessible outside of the class"